src/actions/postsActions.js

```diff
@@ -53,21 +53,11 @@ function getPostsAction(posts: Posts): GetPostsAction {
   };
 }
 
-function getIndex(postsArray, id) {
-  const posts = cloneObject(postsArray);
-  return posts.findIndex(obj => parseInt(obj.id) === parseInt(id));
-}
-
-function cloneObject(postsArray) {
-  return [...postsArray];
-}
-
 export function addPost(post: Post): ThunkAction {
   return dispatch => {
     retrieveData("data", async (err, postsData) => {
```

Comment:
Do you have the current data stored in redux store? If it's so, you don't need to make user wait while getting data from AsyncStorage each time, just use your state
```
return (dispatch, getState) => {
  const data = getState().yourReducer.yourData
```
without `retrieveData`